### PR TITLE
Latency e2e: Report relevant error message upon invalid values

### PR DESCRIPTION
**Pull request by shajmakh**
Latency tests now display to the user a relevant error message about the invalid value that was set to the environment variable, along with the expected to be a valid value.

### CODE REVIEW

File: functests/4_latency/latency.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ const (
 	defaultTestRuntime  = "300"
 	defaultMaxLatency   = -1
 	defaultTestCpus     = -1
+	MAX_INT             = 2147483647
```
Comment (Tal-or) [inline functests/4_latency/latency.go:43]: Capital letters are usually for environment variables.
please write it as ```maxInt```

File: functests/4_latency/latency.go [hunk 2]
```diff
@@ -249,22 +251,36 @@ var _ = Describe("[performance] Latency Test", func() {
 
 func getLatencyTestRun() (bool, error) {
 	if latencyTestRunEnv, ok := os.LookupEnv("LATENCY_TEST_RUN"); ok {
-		return strconv.ParseBool(latencyTestRunEnv)
+		if val, err := strconv.ParseBool(latencyTestRunEnv); err != nil {
+			return val, fmt.Errorf("the environment variable LATENCY_TEST_RUN has incorrect value %q: %w", latencyTestRunEnv, err)
+		} else {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:256]: You don't need the else here, just return.

File: functests/4_latency/latency.go [hunk 3]
```diff
@@ -249,22 +251,36 @@ var _ = Describe("[performance] Latency Test", func() {
...
+		}
 	}
 	return defaultTestRun, nil
 }
 
-func getLatencyTestRuntime() string {
+func getLatencyTestRuntime() (string, error) {
 	if latencyTestRuntimeEnv, ok := os.LookupEnv("LATENCY_TEST_RUNTIME"); ok {
-		return latencyTestRuntimeEnv
+		if val, err := strconv.Atoi(latencyTestRuntimeEnv); err != nil || val < 1 {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:265]: The logic here is a bit confusing, I would separate that, so:
```
val, err := strconv.Atoi(latencyTestRuntimeEnv)
if err != nil {
    return latencyTestRuntimeEnv, fmt.Errorf("the environment variable LATENCY_TEST_RUNTIME has incorrect value %q, it must be a positive integer with maximum value of %d", latencyTestRuntimeEnv, MAX_INT)
  }
  
if val < 1 {
    return "", fmt.Errorf("the environment variable LATENCY_TEST_RUNTIME has a nonpositive value %q, it must be a positive integer with maximum value of %d", latencyTestRuntimeEnv, MAX_INT)
}

return latencyTestRuntimeEnv, nil
```

File: functests/4_latency/latency.go [hunk 4]
```diff
@@ -249,22 +251,36 @@ var _ = Describe("[performance] Latency Test", func() {
...
 	}
-	return defaultTestRuntime
+	return defaultTestRuntime, nil
 }
 
 func getLatencyTestDelay() (int, error) {
 	if latencyTestDelayEnv, ok := os.LookupEnv("LATENCY_TEST_DELAY"); ok {
-		if val, err := strconv.Atoi(latencyTestDelayEnv); err != nil {
-			return val, fmt.Errorf("the environment variable LATENCY_TEST_DELAY has incorrect value %q: %w", latencyTestDelayEnv, err)
+		if val, err := strconv.Atoi(latencyTestDelayEnv); err != nil || val < 0 {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:279]: Same here. Separate the errors

File: functests/4_latency/latency.go [hunk 5]
```diff
@@ -274,8 +290,11 @@ func getLatencyTestDelay() (int, error) {
 
 func getLatencyTestCpus() (int, error) {
 	if latencyTestCpusEnv, ok := os.LookupEnv("LATENCY_TEST_CPUS"); ok {
-		if val, err := strconv.Atoi(latencyTestCpusEnv); err != nil {
-			return val, fmt.Errorf("the environment variable LATENCY_TEST_CPUS has incorrect value %q: %w", latencyTestCpusEnv, err)
+		if val, err := strconv.Atoi(latencyTestCpusEnv); err != nil || val < 1 {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:293]: ditto

File: functests/4_latency/latency.go [hunk 6]
```diff
@@ -292,17 +311,22 @@ func getMaximumLatency(testName string) (int, error) {
 	var err error
 	val := defaultMaxLatency
 	if unifiedMaxLatencyEnv, ok := os.LookupEnv("MAXIMUM_LATENCY"); ok {
-		if val, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil {
-			err = fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q: %w", unifiedMaxLatencyEnv, err)
-			return val, err
+		if val, err = strconv.Atoi(unifiedMaxLatencyEnv); err != nil || val < 0 {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:314]: ditto

File: functests/4_latency/latency.go [hunk 7]
```diff
@@ -292,17 +311,22 @@ func getMaximumLatency(testName string) (int, error) {
...
+			return val, fmt.Errorf("the environment variable MAXIMUM_LATENCY has incorrect value %q, it must be a non-negative integer with maximum value of %d: %w", unifiedMaxLatencyEnv, MAX_INT, err)
 		}
 	}
 
 	// specific values will have precedence over the general one
 	envVariableName := fmt.Sprintf("%s_MAXIMUM_LATENCY", strings.ToUpper(testName))
 	if maximumLatencyEnv, ok := os.LookupEnv(envVariableName); ok {
-		if val, err = strconv.Atoi(maximumLatencyEnv); err != nil {
-			err = fmt.Errorf("the environment variable %q has incorrect value %q: %w", envVariableName, maximumLatencyEnv, err)
+		if val, err = strconv.Atoi(maximumLatencyEnv); err != nil || val < 0 {
```
Comment (Tal-or) [inline functests/4_latency/latency.go:325]: ditto

File: functests/4_latency/latency.go (hunk 1 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:43]: fixed , thanks !

File: functests/4_latency/latency.go (hunk 2 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:256]: my bad , thanks !

File: functests/4_latency/latency.go (hunk 4 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:279]: done , thanks!

File: functests/4_latency/latency.go (hunk 6 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:314]: done , thanks!

File: functests/4_latency/latency.go (hunk 5 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:293]: done , thanks!

File: functests/4_latency/latency.go (hunk 3 above)
Comment (shajmakh) [reply functests/4_latency/latency.go:265]: done , thanks!

File: functests/4_latency/latency.go [hunk 8]
```diff
@@ -40,6 +40,7 @@ const (
 	defaultTestRuntime  = "300"
 	defaultMaxLatency   = -1
 	defaultTestCpus     = -1
+	maxInt              = 2147483647
```
[the remark this answers is not in the document]
Comment (shajmakh) [reply functests/4_latency/latency.go:43]: my bad, in another place I used it as a string, will change it , Thanks !

File: functests/4_latency/latency.go [hunk 9]
```diff
@@ -40,6 +41,7 @@ const (
 	defaultTestRuntime  = "300"
 	defaultMaxLatency   = -1
 	defaultTestCpus     = -1
+	maxInt              = math.MaxInt32
```
Comment (cynepco3hahue) [inline functests/4_latency/latency.go:44]: you do not need it under the new const, it already const and you can use it straight forward :)

Comment (shajmakh) [reply functests/4_latency/latency.go:44]: uh my mistake , correcting it , thanks !

**Pull request merged**